runtime/v1/linux/proc/exec_state.go

```diff
@@ -146,40 +127,30 @@ type execStoppedState struct {
 func (s *execStoppedState) transition(name string) error {
 	switch name {
 	case "deleted":
-		s.p.State = &deletedState{}
+		s.p.execState = &deletedState{}
 	default:
 		return errors.Errorf("invalid state transition %q to %q", stateName(s), name)
 	}
 	return nil
 }
 
 func (s *execStoppedState) Resize(ws console.WinSize) error {
-	s.p.mu.Lock()
-	defer s.p.mu.Unlock()
-
 	return errors.Errorf("cannot resize a stopped container")
 }
 
 func (s *execStoppedState) Start(ctx context.Context) error {
-	s.p.mu.Lock()
-	defer s.p.mu.Unlock()
-
 	return errors.Errorf("cannot start a stopped process")
 }
 
 func (s *execStoppedState) Delete(ctx context.Context) error {
 	if err := s.p.delete(ctx); err != nil {
 		return err
 	}
-	s.p.mu.Lock()
```

Comment:
Just point out that the lock location is changed in this PR. The lock was put here in https://github.com/containerd/containerd/pull/2757.

But I think the move is fine, because grabbing process lock only potentially blocks operations to the process. It is fine, as long as it doesn't block the whole shim. Actually the other fix https://github.com/containerd/containerd/pull/2757 in is more important.

Reply:
ok, let me think about it some

Reply:
I think the change made by this PR is fine. Not grabbing shim lock is more important. :)